apps/balancer-tools/src/app/apr/round/(components)/TokenFilterInput.tsx

```diff
@@ -0,0 +1,51 @@
+"use client";
+
+import { useSearchParams } from "next/navigation";
+
+import { POOLS_WITH_LIVE_GAUGES } from "#/lib/balancer/gauges";
+
+import { MultiSelectDropdown } from "./MultiSelectDropdown";
+
+const AVALIABLE_TOKENS = [
+  ...new Set(
+    POOLS_WITH_LIVE_GAUGES.flatMap((pool) =>
+      pool.tokens.map((token) => token.symbol),
+    ),
+  ),
+];
+
+export function TokenFilterInput() {
+  const searchParams = useSearchParams();
+
+  const handleTokenSelect = (selectedItems: string[]) => {
+    const current = new URLSearchParams(Array.from(searchParams.entries())); // -> has to use this form
+
+    if (!selectedItems.length) {
+      current.delete("tokens");
+    } else {
+      current.set("tokens", selectedItems.join("_"));
+    }
+
+    const search = current.toString();
+    const query = search ? `?${search}` : "";
+
+    // Workaround while Next.js doesn't support shallow routing
+    // https://github.com/vercel/next.js/discussions/48110
+    window.history.pushState({}, "", window.location.pathname + query);
```

Comment:
Hey @ribeirojose @luizakp, I spent some time trying to get this play nicely, but I can't find a proper workaround  for this.

<details><summary>Code as it is in this commit</summary>
It updates the URL but wont make components change (no refresh)

https://github.com/bleu-studio/balancer-tools/assets/68453900/12c933a8-63eb-4f3c-ada2-a8bba73a0657


</details>


<details><summary>using router.push</summary>
It updates components, but makes the page scroll up

https://github.com/bleu-studio/balancer-tools/assets/68453900/5992df7d-9a88-4f1f-a5ed-1beb847944a7

</details>

In next 12 it was prety easy due by using [shallow routing](https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating#shallow-routing), but it currently isnt implemented on the router.

My solution will be by using Context so we can share with the components, what do you guys think?

Reply:
the goal here is to change the url as you are applying the filter?

Reply:
Pretty much, it should change the query param, so `?tokens=USDC,qETH` to `?tokens=USDC` vice versa